ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -255,13 +252,13 @@ private void handleDatePickerInteraction() {
   }
 
   /**
-   * Handle checkbox interaction to control if the date picker is enabled or not.
+   * Binds the enabled status of the start and end date pickers with the checkbox selected property.
    */
   @FXML
   private void handleCheckBoxInteraction() {
 
-    startDatePicker.setDisable(!timeExtentCheckBox.isSelected());
-    endDatePicker.setDisable(!timeExtentCheckBox.isSelected());
+    startDatePicker.disableProperty().bind(timeExtentCheckBox.selectedProperty().not());
```

Comment:
Only thing I can think to check here is if a bidirectional binding would be better - that way if the date picker was disabled the checkbox would become disabled. Probably not necessary in your UI but a handy feature to keep enablement in sync just in case the date picker is programmatically disabled.